client.go

```diff
@@ -346,6 +363,34 @@ func WithOAuth() ClientOptions {
 	}
 }
 
+func WithScopedOAuthApp(clientId string, clientSecret string, scopes []string) ClientOptions {
+	ts := baseTokenSource(context.Background(), clientId, clientSecret, scopes)
```

Comment:
We should probably remove `context.Background()` and pass it in as a parameter to `WithScopedOAuthApp`